crates/bevy_reflect/src/enums/dynamic_enum.rs

```diff
@@ -135,13 +135,13 @@ impl DynamicEnum {
     }
 
     /// Set the current enum variant represented by this struct.
-    pub fn set_variant<I: Into<String>, V: Into<DynamicVariant>>(&mut self, name: I, variant: V) {
+    pub fn set_variant<I: Into<&'static str>, V: Into<DynamicVariant>>(&mut self, name: I, variant: V) {
```

Comment:
One concern is: how would a user generate these dynamically? 

For example, they might have their own custom deserialization logic which returns a `String`. Or maybe they're parsing a script that contains non-Rust enum types.

They _could_ intern their strings, but I'm not sure we want to require that until we have a better idea of how we might want to handle these dynamic cases. And I have a feeling we won't be able to do that until we see Unique Reflect in action.

Reply:
One approach would be to leak the memory of the string? Set_variant shouldn't be called very often.

I guess it's problematic to return `&'static str` for variant_names in the DynamicEnum case..